backend/src/index.spec.ts

```diff
@@ -1,28 +1,52 @@
 import { resolvers } from './resolvers'
 import { typeDefs } from './typeDefs'
+import { permissions } from './permissions'
 import { PostsDataSource } from './postsDataSource'
 import { ApolloServer } from 'apollo-server'
 import { createTestClient } from 'apollo-server-testing'
+import { applyMiddleware } from 'graphql-middleware'
+import { makeExecutableSchema } from 'graphql-tools'
+const bcrypt = require('bcrypt')
+const jwt = require('jsonwebtoken')
+require('dotenv').config()
+
+const testContext = (testToken?) => {
+  let token = testToken || ''
+  token = token.replace('Bearer ', '')
+  try {
+    const decodedJwt = jwt.verify(
+      token,
+      process.env.JWTSECRET
+    )
+    return { decodedJwt }
+  } catch (e) {
+    return {}
+  }
+}
+const setupServer = (postData, userData, testToken?) => {
+  const schema = makeExecutableSchema({ typeDefs, resolvers })
+  const server = new ApolloServer({
+    schema: applyMiddleware(schema, permissions),
+    context: testContext(testToken), // not sure if this is the correct way. but we didn`t find another solution to add the token as request (see https://github.com/apollographql/apollo-server/issues/2277)
+    dataSources: () => ({
+      posts: new PostsDataSource(postData, userData)
+    })
+  })
+  return createTestClient(server)
+}
 
 describe('Test apollo server queries', () => {
   it('get all posts returns 1 post', async () => {
     const postData = [
       { id: 'post1', title: 'Item 1', votes: 0, voters: [], author: {} }
     ]
     const userData = [
-      { name: 'user1', posts: [postData[0]] }
+      { id: 'userid1', name: 'user1', email: 'user1@example.org', password: bcrypt.hashSync('user1password', 10), posts: [postData[0]] }
```

Comment:
Do you have a JS class `User`? An object-oriented approach would be to call ` bcrypt.hashSync` in the constructor and  implement a instance method called `comparePassword(password)` which compares against `this.hashedPassword`.